spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -655,8 +655,23 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
           (builder, mathExpr) => builder.setIntegralDivide(mathExpr))
 
         if (divideExpr.isDefined) {
+          val childExpr = if (dataType.isInstanceOf[DecimalType]) {
+            // check overflow for decimal type
+            val builder = ExprOuterClass.CheckOverflow.newBuilder()
+            builder.setChild(divideExpr.get)
+            builder.setFailOnError(false)
+            builder.setDatatype(serializeDataType(dataType).get)
+            Some(
+              ExprOuterClass.Expr
+                .newBuilder()
+                .setCheckOverflow(builder)
+                .build())
+          } else {
+            divideExpr
+          }
+
           // cast result to long
-          castToProto(expr, None, LongType, divideExpr.get, CometEvalMode.LEGACY)
+          castToProto(expr, None, LongType, childExpr.get, CometEvalMode.LEGACY)
```

Comment:
Could be just naming issue but why it is `child`?

Reply:
> Could be just naming issue but why it is `child`?

It is for castToLong that is `child`, and it could be divide expr or checkoverflow expor so I can't think of a better name, do you have any suggestions?